Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Pass/PassContainer.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+// This header file is for declaring types used for RPI System classes to avoid recursive includes
+ 
+#include <Atom/RPI.Public/Base.h>
+#include <Atom/RPI.Public/Pass/Pass.h>
+#include <Atom/RPI.Public/Pass/ParentPass.h>
+
+namespace AZ::RPI
+{
+    // Helper class used by the PassSystem and RenderPipeline to contain and update passes
+    // Passes owned by the container are stored as a tree under the container's root pass
+    // The container has queues for pass building, initialization and removal. These queues
+    // are so that logic modifying or removing passes isn't triggered while the passes are 
+    // rendering, but instead at the start of the frame when it is safe to do so.
+    class PassContainer
```

Comment:
nit: this isn't really a pass container, it's tree update wrapper. If it was a container it would hold more than 1 item.

Reply:
It also holds that tree, which makes it a container, no?
Just as you can have a linked list that just points to the first element in the list, you can have a collection of passes and just point to the root of the tree.
Or do you feel container means something else that this could be confused with? Open to suggestions for better names, but prefer PassContainer over PassTreeUpdateWrapper. And it does perform the function of a container in that the Pipeline owns a bunch of passes, and it holds ownership of those passes through this class.

Reply:
PassTree?